File: app/src/main/java/mohtasham/paydar/sabalan/ezbazi/model/RentType.java
package mohtasham.paydar.sabalan.ezbazi.model;

import org.json.JSONException;
import org.json.JSONObject;

public class RentType {
  private int id;
  private int day_count;
  private int price_percent;

  public static class Parser{
    public static RentType parse(JSONObject object){
      RentType rentType = new RentType();
      try {
        rentType.setId(object.getInt("id"));
        rentType.setDay_count(object.getInt("day_count"));
        rentType.setPrice_percent(object.getInt("price_percent"));
      } catch (JSONException e) {
        e.printStackTrace();
      }
      return rentType;
    }

  }

  public int getId() {
    return id;
  }

  public void setId(int id) {
    this.id = id;
  }

  public int getDay_count() {
    return day_count;
  }

  public void setDay_count(int day_count) {
    this.day_count = day_count;
  }

  public int getPrice_percent() {
    return price_percent;
  }

  public void setPrice_percent(int price_percent) {
    this.price_percent = price_percent;
  }
}

File: app/src/main/java/ir/easybazi/model/RentType.kt
package ir.easybazi.model

import org.json.JSONException
import org.json.JSONObject

import ir.easybazi.system.application.G

class RentType {
    var id: Int = 0
    var day_count: Int = 0
    var price_percent: Int = 0

    object Parser {
        fun parse(`object`: JSONObject): RentType {
            val rentType = RentType()
            try {
                rentType.id = `object`.getInt("id")
                rentType.day_count = `object`.getInt("day_count")
                rentType.price_percent = `object`.getInt("price_percent")
            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return rentType
        }

    }

    companion object {


        fun findById(id: Int): RentType {
            var rentType = RentType()
            for (i in G.rentTypes!!.indices) {
                if (id == G.rentTypes!![i].id) {
                    rentType = G.rentTypes!![i]
                    break
                }
            }
            return rentType
        }
    }
}
